sympy/functions/combinatorial/numbers.py

```diff
@@ -1858,12 +1858,37 @@ def stirling(n, k, d=None, kind=2, signed=False):
 
 
 @cacheit
-def _nT(n, k):
+def _nT(*args):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
-    if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+    if len(args) == 2:
+        n = args[0]
+        k = args[1]
+        if k > n:
+            return 0
+        if k >= n/2.0:
+            args = [n - k]
+    numparts = 0
+    if len(args) == 1:
+        n = args[0]
+        numparts = partition(n)
+
+    elif len(args) == 2:
+        n = args[0]
+        k = args[1]
+        if n == k or k == 1:
+            numparts = 1
+        elif n < k or k == 0:
+            numparts = 0
+        else:
+            q1 = int(n)
```

Comment:
What is the purpose of the temporary variable here? If you eliminate this and keep the n, k arguments and assume they are ints already (since this is a private routine and the calling routine already handled that conversion) you should be able to reduce the code to half the number of current lines...or better.

Reply:
Okay i will eliminate q1 and  k1